src/ftpclass.cc

```diff
@@ -1535,7 +1535,7 @@ int   Ftp::Do()
 	    }
 	 }
 #if USE_SSL
-	 if(strcmp(proxy_auth_type,"joined") || strcmp(proxy_auth_type,"joined-acct"))
+	 if(strcmp(proxy_auth_type,"joined") != 0 && strcmp(proxy_auth_type,"joined-acct") != 0)
```

Comment:
I added a "auth_allowed" flag to the proxy things. I think:

```
user && pass
proxy_user && proxy_pass
```

Need to be adjusted in another pull request. I think general refactoring might be needed, regarding those often used conditions. Maybe some flag like "has_upw" and "has_proxy_upw".